distributed/worker.py

```diff
@@ -1601,7 +1601,10 @@ def _close():
             # weird deadlocks particularly if the task that is executing in
             # the thread is waiting for a server reply, e.g. when using
             # worker clients, semaphores, etc.
-            await to_thread(_close)
+            try:
+                await to_thread(_close)
+            except RuntimeError:  # Are we shutting down the process?
+                _close()  # Just run it directly
```

Comment:
I have a slight preference to be explicit here but I'm fine either way. The following works as well, at least for the test

```suggestion
            from distributed.utils import is_python_shutting_down
            if is_python_shutting_down():
                # Scheduling new futures during shutdown is not allowed
                _close()
            else:
                await to_thread(_close)
```

Reply:
My preference is to ask for forgiveness rather than permission here.  I don't have enough confidence about Python shutting down to know all of the stages involved.  Honestly I'd also be ok just `except Exception` here.  The worst thing that can happen is that we call close again.  I'd rather take the sledgehammer than the scalpel approach here.